pkg/controller/epgcache.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2021 Cisco Systems, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package controller
+
+import (
+	"github.com/noironetworks/aci-containers/pkg/apicapi"
+	"github.com/noironetworks/aci-containers/pkg/metadata"
+
+	"encoding/json"
+	"fmt"
+	"sort"
+	"strings"
+)
+
+func (cont *AciController) clearFaultInstances() {
+
+	queryfilter := fmt.Sprintf("query-target-filter=and(wcard(vmmClusterFaultInfo.dn,\"comp/prov-%s/ctrlr-\\[%s\\]-%s\"))",
+		cont.vmmDomainProvider(), cont.config.AciVmmDomain, cont.config.AciVmmDomain)
+	args := []string{
+		queryfilter,
+		"order-by=vmmClusterFaultInfo.modTs|desc",
+	}
+
+	faultUri := fmt.Sprintf("/api/node/class/vmmClusterFaultInfo.json?%s", strings.Join(args, "&"))
+	apicresp, err := cont.apicConn.GetApicResponse(faultUri)
+	if err != nil {
+		return
+	}
+	for _, obj := range apicresp.Imdata {
+		for _, body := range obj {
+			dn, ok := body.Attributes["dn"].(string)
+			if ok {
+				cont.indexMutex.Lock()
+				cont.apicConn.DeleteDn(dn)
+				cont.log.Debug("Clearing existing Fault Instance: ", dn)
+				cont.indexMutex.Unlock()
+			}
+		}
+	}
+	return
+}
+
+func (cont *AciController) getSubString(dn string) string {
```

Comment:
can you rename to something like getEpgFromEppd or something to make it clearer what the function is doing?

Reply:
Okay will do